plugins/BassBooster/BassBooster.cpp

```diff
@@ -83,11 +83,18 @@ bool BassBoosterEffect::processAudioBuffer( sampleFrame* buf, const fpp_t frames
 	if( m_bbControls.m_gainModel.isValueChanged() ) { changeGain(); }
 	if( m_bbControls.m_ratioModel.isValueChanged() ) { changeRatio(); }
 
+	float gain = m_bbControls.m_gainModel.value();
+	ValueBuffer *gainBuffer = m_bbControls.m_gainModel.valueBuffer();
+	int gainInc = gainBuffer ? 1 : 0;
+	float *gainPtr = gainBuffer ? &( gainBuffer->values()[ 0 ] ) : &gain;
+
 	double outSum = 0.0;
 	const float d = dryLevel();
 	const float w = wetLevel();
 	for( fpp_t f = 0; f < frames; ++f )
 	{
+		m_bbFX.leftFX().setGain( *gainPtr );
+		m_bbFX.rightFX().setGain( *gainPtr );
 		outSum += buf[f][0]*buf[f][0] + buf[f][1]*buf[f][1];
```

Comment:
hmm, this setGain call causes a lot of overhead since it's a function call. I think in this case it'd be better to use two separate for loops, since we only have one parameter we want to make sample-exact.